src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp

```diff
@@ -0,0 +1,234 @@
+        return;
+    }
+
+    std::ifstream file(fileName);
+
+    if (!file.is_open())
+    {
+        // std::cout << "couldn't open sudoers file: " << fileName << std::endl;
+        return;
+    }
+
+    bool isLongLine = false;
+    auto contents = std::string((std::istreambuf_iterator<char>(file)),
+                                std::istreambuf_iterator<char>());
+    auto lines{Utils::split(contents, '\n')};
+
+    for (auto& line : lines)
+    {
+        // sudoers uses EBNF for grammar.
+        trim(line);
+
+        if (line.empty())
+        {
+            continue;
+        }
+
+        // if last line contains a backslash as the last character, treat current line as part
+        // of previous line and append it to appropriate column.
+        if (isLongLine)
+        {
+            isLongLine = (line.back() == '\\');
+            auto& lastLine = results.back();
+
+            if (lastLine["rule_details"].empty())
+            {
+                if (lastLine["header"].empty())
+                {
+                    std::string tmp = lastLine["header"];
+                    lastLine["header"] = tmp;
+                }
+            }
+            else
+            {
+                if (lastLine["rule_details"].empty())
+                {
+                    std::string tmp = lastLine["rule_details"];
+                    lastLine["rule_details"] = tmp;
+                }
+            }
+
+            lastLine["rule_details"] = lastLine["rule_details"].get<std::string>() + line;
+            continue;
+        }
+
+
+        // Find the rule header.
+        auto headerLen = line.find_first_of("\t\v ");
+        auto header = line.substr(0, headerLen);
+        trim(header);
+
+        // We frequently check if these are include headers. Do it once here.
+        auto isInclude = (header == "#include" || header == "@include");
+        auto isIncludeDir = (header == "#includedir" || header == "@includedir");
+
+        // skip comments.
+        if (line.at(0) == '#' && !isInclude && !isIncludeDir)
+        {
+            continue;
+        }
+
+        // Find the next field. Instead of skipping the whitespace, we
+        // include it, and then trim it.
+        auto ruleDetails = (headerLen < line.size()) ? line.substr(headerLen) : "";
+        trim(ruleDetails);
+
+        // If an include is _missing_ the target to include, treat it like a comment.
+        if (ruleDetails.empty() && (isInclude || isIncludeDir))
+        {
+            continue;
+        }
+
+        // Check if a blackslash is the last character on this line.
+        if (!isInclude && !isIncludeDir && line.back() == '\\')
+        {
+            isLongLine = true;
+        }
+
+        nlohmann::json entry;
+        entry["header"] = header;
+        entry["source"] = fileName;
+        entry["rule_details"] = ruleDetails;
+        results.push_back(std::move(entry));
+
+        auto resolvePath = [&](const std::string & relativePath)
```

Comment:
Same. Maybe in the file `shared_modules/utils/fileSystem.h` or `shared_modules/utils/filesystemHelper.h`

Reply:
Done!